### PR TITLE
Admin page UI

### PR DESCRIPTION
**Pull request by OleanjiKingCode**
# Admin page UI Creation

- Wiki created/edited graph  ✅  
- Metacards for  created/edited/total editors and visitors card  ✅  
- Table to display unarchived wikis ✅   
- Table to display active and banned users ✅   

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/568

### CODE REVIEW

File: src/components/Admin/UserDataPie.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      <Heading as="h2" fontSize="21" fontWeight="bold" w="full">
+        User Data
+      </Heading>
+      <PieChart width={350} height={400}>
+        <Pie
+          data={piedata}
+          cx={170}
+          cy={200}
+          innerRadius={50}
+          outerRadius={130}
```
Comment (anubra266) [inline src/components/Admin/UserDataPie.tsx:25]: You can remove this, if you want the chart to fill its container

**Comment by kesar** [comment]
its looking very promising. I would suggest you wrap the whole content to make sure its centered on big screens (that way layout looks better) check what's the wrapper on about page 

image

image

**Comment by kesar** [comment]
for getting data check the graph doc or talk to ashakae.

for getting totals for the first row, consider using something like:

{
  wikisEdited(startDate: 0, endDate: 1658828593 (today), interval: "year") {
    amount
  }
}

{
  wikisCreated(startDate: 0, endDate: 1658828593 (today), interval: "year") {
    amount
  }
}

and for the graph w lines, you can also use those queries switching params. We dont have data for visitors for now.

**Comment by kesar** [comment]
also remember you need to implement login system similar to settings

**Comment by kesar** [comment]
I briefly checked this branch and it looks its still in very early stage, I was expecting this to be more complete giving 2 ppl 17 days already. I haven't seen any code change in last 3 days from any of you neither.

From main specs: 
done - list of created wikis ordered by last created, 
not done - with options to hide
not done - or promote to homepage. 
not done - it should have a search on top of the wiki table. 
not done - it should allow to filter to show only promoted ones

done - list of users and 
not done - possibility to block them.

not done - chart with number of wikis created / wikis edited per day / week / month

so out of all specs only the stats and 2 listings seems to be complete, missing all the functionalities to promote , hide, search, filters... API for search or hide / promote wikis has been done since 2 weeks ago.

**Comment by kesar** [comment]
fyi: sorting its not part of the specs of the issue. And if you implement sorting, it needs to be at API level. There is no reason for sorting a column with a subset of data.

Please make sure you read the specs properly and focus in the most important use cases.

**Comment by OleanjiKingCode** [comment]
> fyi: sorting its not part of the specs of the issue. And if you implement sorting, it needs to be at API level. There is no reason for sorting a column with a subset of data.
> 
> Please make sure you read the specs properly and focus in the most important use cases.



> fyi: sorting its not part of the specs of the issue. And if you implement sorting, it needs to be at API level. There is no reason for sorting a column with a subset of data.
> 
> Please make sure you read the specs properly an

**Comment by OleanjiKingCode** [comment]
> fyi: sorting its not part of the specs of the issue. And if you implement sorting, it needs to be at API level. There is no reason for sorting a column with a subset of data.
> 
> Please make sure you read the specs properly and focus in the most important use cases.

ohh i didnt know and i just spent the last hour doing it

**Review by Simplypheyie96 — COMMENTED** [review]
-padding in this button is a little too big now.
image
I think a lot of things are yet to be implemented based on what we talked about today. We can have a follow up meeting to get more things in persperctive.

**Comment by OleanjiKingCode** [comment]
> not done - or promote to homepage.

i can use what ive done as a template for when the api is ready

**Comment by kesar** [comment]
promote and hide APIs are ready since 2 weeks ago if I am not wrong.

**Comment by Softdev1** [comment]
Yes, you are very right @kesar, the API has been provisioned by ashaka

**Review by Softdev1 — COMMENTED** [review]
Well done but there are some things in the designs that are not implemented here properly Please check the Figma properly and ensure it is implemented in the right way. focus more on the top priority things for now and leave sorting.

File: src/pages/account/settings.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React, { useEffect, useState } from 'react'
+import React, { useEffect } from 'react'
```
Comment (kesar) [inline src/pages/account/settings.tsx:1]:
```suggestion
import React, { useEffect, useState } from 'react'
```

File: src/pages/account/settings.tsx [hunk 3]
```diff
@@ -37,10 +37,6 @@ const Settings = () => {
     }
   }, [userAddress, setAccount, token])
```
Comment (kesar) [inline src/pages/account/settings.tsx:39]:
```suggestion

  const [mounted, setMounted] = useState(false)
  useEffect(() => setMounted(true), [])
  if (!mounted) return <></>
```

**Review by kesar — APPROVED** [review]
lets merge and open new pr w improvements

**Pull request merged**